go.mod

```diff
@@ -123,3 +123,5 @@ require (
 	sigs.k8s.io/json v0.0.0-20241010143419-9aa6b5e7a4b3 // indirect
 	sigs.k8s.io/structured-merge-diff/v4 v4.4.2 // indirect
 )
+
+replace sigs.k8s.io/karpenter => sigs.k8s.io/karpenter v1.4.1-0.20250506155914-1c76d2b06e41
```

Comment:
Can we drop this? I think drop this and rebase from main. This should have your changes